packages/pds/src/api/com/atproto/repo/importRepo.ts

```diff
@@ -0,0 +1,128 @@
+import { Server } from '../../../../lexicon'
+import AppContext from '../../../../context'
+import { ActorStoreTransactor } from '../../../../actor-store'
+import { TID } from '@atproto/common'
+import {
+  Repo,
+  WriteOpAction,
+  getAndParseRecord,
+  readCarStream,
+  verifyDiff,
+} from '@atproto/repo'
+import { InvalidRequestError } from '@atproto/xrpc-server'
+import { CID } from 'multiformats/cid'
+import PQueue from 'p-queue'
+import { AtUri } from '@atproto/syntax'
+import { BlobRef, LexValue, RepoRecord } from '@atproto/lexicon'
+
+export default function (server: Server, ctx: AppContext) {
+  server.com.atproto.repo.importRepo({
+    auth: ctx.authVerifier.accessNotAppPassword,
+    handler: async ({ input, auth }) => {
+      const did = auth.credentials.did
+      if (!ctx.cfg.service.acceptingImports) {
+        throw new InvalidRequestError('Service is not accepting repo imports')
+      }
+      await ctx.actorStore.transact(did, (store) =>
+        importRepo(store, input.body),
+      )
```

Comment:
I do think that in general—i.e. even if not given our implementation/policy—we can expect this to last long enough that it might time out the connection.  And some implementations might want to plop it on a job queue and deal with it totally asynchronously.

Even we might consider just tossing the repo car on disk or into s3 then working with it streaming or in parts to make it robust to repos that don't happily fit into memory.

Reply:
(To be clear I think this is fine for our current purposes, mostly want to leave the door open to these possibilities.)

Reply:
yeah hmmm. is there a way that we can do both? like process it synchronously in most cases, but drop into async if it's large enough? Not sure how that would surface in schemas 🤔